grudge/execution.py

```diff
@@ -73,7 +73,15 @@ def map_ones(self, expr):
 
     def map_node_coordinate_component(self, expr):
         discr = self.dcoll.discr_from_dd(expr.dd)
-        return thaw(self.array_context, discr.nodes()[expr.axis])
+        return thaw(self.array_context, discr.nodes(
+            # only save volume nodes or boundary nodes
+            # (but not nodes for interior face discretizations, which are likely only
+            # used once to compute the normals)
```

Comment:
Ah okay!

Though I wondering if this will be an issue for entropy stable DG. For example, when doing the volume flux differencing, we need to compute values on the surface of elements too. Which means we have to interpolate to the face nodes. But I guess if those interpolations are already done, there's no need to hang on to the surface nodes though? So maybe it's okay

Reply:
This all deals with a global node array (hence the memory consumption concern). I don't believe those are used in constructing the interpolation (just uses reference nodes).

Reply:
OK, thanks!